src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,34 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private final Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int randomChoice = random.nextInt(5);
+
+        switch (randomChoice) {
+            case 0:
+                return new Square(random.nextDouble() * 10, colorSupplier.getRandomColor().name());
```

Comment:
lets add private methods to get each random figure
```suggestion
            case 0:
                return getRandomSquare();
```